article_store/articles/api.py

```diff
@@ -0,0 +1,206 @@
+from lxml.etree import (
+    Element,
+    SubElement,
+    tostring,
+)
+from django.db.models import QuerySet
+from django.http.response import HttpResponse
+from django.core.exceptions import ObjectDoesNotExist, ValidationError
+from rest_framework import status, viewsets
+from rest_framework.request import Request
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from rest_framework_xml.renderers import XMLRenderer
+
+from articles.xml_parser import ArticleXMLParser
+
+from articles.models import (
+    Article,
+    ArticleVersion,
+    Content,
+)
+from articles.serializers import (
+    ArticleSerializer,
+    ArticleVersionSerializer,
+)
+
+from articles.utils import parse_accept_language_header
+
+
+class ArticleViewSet(viewsets.ModelViewSet):
+    model = Article
+    queryset = Article.objects.all()
+    serializer_class = ArticleSerializer
+
+
+def article_list_xml_generator(articles: 'QuerySet[Article]') -> Element:
+    """Generate an xml representation of all `Article` objects.
+
+    :param articles: QuerySet[Article]
+    :return: class: `Element`
+    """
+    name_space = {'libero': 'http://libero.pub'}
+
+    root = Element('{%s}articles' % name_space['libero'], nsmap=name_space)
+
+    for article in articles:
+        child = SubElement(root, '{%s}article' % name_space['libero'])
+        child.text = str(article.id)
+
+    return root
+
+
+class ArticleXMLAPIView(APIView):
+    parser_classes = (ArticleXMLParser,)
+    renderer_classes = (XMLRenderer,)
+
+    default_part_name = 'front'
+    latest = 'latest'
+    style_element = '<?xml version="1.0" ?>\n'
+
+    def post(self, request: Request, article_id: str) -> HttpResponse:
+        """
+
+        :param request: class: `Request`
+        :param article_id: str
+        :return: class: `HttpResponse`
+        """
+
+        if not article_id or not Article.id_is_valid(article_id):
+            return Response({'error': 'Please provide a valid article id'}, status=status.HTTP_406_NOT_ACCEPTABLE)
+
+        article, created = Article.objects.get_or_create(id=article_id)
+        article_version = ArticleVersion.objects.create(version=article.next_version,
+                                                        article=article)
+
+        for content_item in request.data.get('content-list', []):
+            Content.objects.create(article_version=article_version, **content_item)
```

Comment:
are there transaction boundaries here? I think at least the `ArticleVersion` and the `Content` objects should be inside the same database transaction